src/microbe_stage/editor/MicrobePartSelection.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System.Globalization;
+using Godot;
+
+/// <summary>
+///   A specialized button to display a microbe part for selection in the cell editor.
+/// </summary>
+public class MicrobePartSelection : MarginContainer
+{
+    [Export]
+    public ButtonGroup SelectionGroup;
+
+    private Label mpLabel;
+    private Button button;
+    private TextureRect iconRect;
+    private Label nameLabel;
+
+    private int mpCost;
+    private Texture partIcon;
+    private string name;
+    private bool locked;
+    private bool selected;
+
+    /// <summary>
+    ///   Emitted whenever the button is selected.
+    /// </summary>
+    [Signal]
+    public delegate void OnPartSelected(string name);
+
+    [Export]
+    public int MPCost
+    {
+        get => mpCost;
+        set
+        {
+            mpCost = value;
+            UpdateLabels();
+        }
+    }
+
+    [Export]
+    public Texture PartIcon
+    {
+        get => partIcon;
+        set
+        {
+            partIcon = value;
+            UpdateIcon();
+        }
+    }
+
+    /// <summary>
+    ///   User readable name. Note that the node Name property should be the "InternalName".
+    /// </summary>
+    [Export]
+    public string PartName
+    {
+        get => name;
+        set
+        {
+            name = value;
+            UpdateLabels();
+        }
+    }
+
+    /// <summary>
+    ///   Currently only makes the button unselectable if true.
+    /// </summary>
+    [Export]
+    public bool Locked
+    {
+        get => locked;
+        set
+        {
+            locked = value;
+
+            UpdateButton();
+            UpdateIcon();
+            UpdateLabels();
+        }
+    }
+
+    public bool Selected
+    {
+        get => selected;
+        set
+        {
+            selected = value;
+
+            UpdateButton();
+            UpdateIcon();
+        }
+    }
+
+    public override void _Ready()
+    {
+        mpLabel = GetNode<Label>("VBoxContainer/MP");
+        button = GetNode<Button>("VBoxContainer/Button");
+        iconRect = GetNode<TextureRect>("VBoxContainer/Button/Icon");
+        nameLabel = GetNode<Label>("VBoxContainer/Name");
+
+        UpdateButton();
+        UpdateLabels();
+        UpdateIcon();
+    }
+
+    public override void _Notification(int what)
+    {
+        if (what == NotificationTranslationChanged)
+        {
+            UpdateLabels();
```

Comment:
One thing to consider regarding this: when translations change, the names of organelles change as well. So anyway that change needs to get passed here, so doesn't this being here just cause duplicate work?

Reply:
That somehow works independently updating the organelle name last time I tested changing the language (without having to reassign `PartName`), but doesn't seem to work anymore now... Will fix this I guess.